topology/probes/istio/virtualservice.go

```diff
@@ -48,28 +48,31 @@ func newVirtualServiceProbe(client interface{}, g *graph.Graph) k8s.Subprobe {
 	return k8s.NewResourceCache(client.(*kiali.IstioClient).GetIstioNetworkingApi(), &kiali.VirtualService{}, "virtualservices", g, &virtualServiceHandler{})
 }
 
+type rule struct {
+	Routes []struct {
+		Destination struct {
+			App     string `mapstructure:"host"`
+			Version string `mapstructure:"subset"`
+		} `mapstructure:"destination"`
+		Weight int `mapstructure:"weight"`
+	} `mapstructure:"route"`
+}
+
 type virtualServiceSpec struct {
-	HTTP []struct {
-		Route []struct {
-			Destination struct {
-				App     string `mapstructure:"host"`
-				Version string `mapstructure:"subset"`
-			} `mapstructure:"destination"`
-			Weight int `mapstructure:"weight"`
-		} `mapstructure:"route"`
-	} `mapstructure:"http"`
+	HTTP []rule `mapstructure:"http"`
+	TLS  []rule `mapstructure:"tls"`
+	TCP  []rule `mapstructure:"tcp"`
 }
 
-func (vsSpec virtualServiceSpec) getAppsVersions() map[string][]string {
-	appsVersions := make(map[string][]string)
-	for _, http := range vsSpec.HTTP {
-		for _, route := range http.Route {
+func getInstances(rules []rule, instances map[string][]string) map[string][]string {
+	for _, r := range rules {
```

Comment:
s/r/rule/ as its' `r` is overloaded in this context (could mean rule or route)

Reply:
DONE